---
permalink: /pages/software
title: Schuyler - Software
---
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <!-- PAGE TITLE -->
        <title>{% if page.title %}{{ page.title }}{% else %}{{ site.data.config.title }}{% endif %}</title>
        <link rel="canonical" href="{{ page.url | replace:'index.html','' | prepend: site.baseurl | prepend: site.url }}">
        <!-- PAGE ICON -->
        {% if site.data.config.page_icon %}<link rel="shortcut icon" href="{{ site.data.config.page_icon }}" />{% endif %}
        <!-- LOAD OPEN SANS FONT -->
        <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700&subset=latin' rel='stylesheet' type='text/css'>
        <!-- LOAD FONTAWESOME ICONS -->
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
        <!-- LOAD ACADEMIC FONTAWESOME STYLE ICONS -->
        <link rel="stylesheet" href="/css/academicons/css/academicons.min.css"/>
        <!-- LOAD LOCAL CSS -->
        <link rel="stylesheet" href="{{ site.baseurl }}{{ '/css/main.css' }}">
    </head>
    <body class="home-page">
        <!-- BACKGROUND IMAGE -->
        <img src="{{ site.data.config.background }}" id="background">
        <!-- NAVIGATION BAR -->
        <div id="navmenu">
            <ul>
                {% for link in site.data.pages %}
                    {% assign class = nil %}
                    {% if page.url == link.url %}
                        <li><a class="active">{{ link.title }}</a></li>
                    {% else %}
                        <li><a href="{{ site.baseurl }}{{ link.url }}">{{ link.title }}</a></li>
                    {% endif %}
                {% endfor %}
            </ul>
        </div>
        <!-- END NAVIGATION BAR -->
        
<!--    PAGE     -->

        <div class="publications" >
        <div id="sw">
        <br>
            <ul>
                <li><font size="+2"><b><a href="https://schuyler-smith.github.io/phylosmith/" target="_blank">phylosmith</a></b></font><br><br>
                <font size="-1">An R-Package hosted on <a href="https://github.com/schuyler-smith/phylosmith/" target="_blank">Github</a>. The package contains functions for reproducible and efficient microbiome analysis with phyloseq-objects. Functions include data-wrangling, graphing, and computational processes designed to make complex R coding into simple one-line functions by exploiting the data-standard of phyloseq, and the redundancies of microbiome datasets.
                <br><br>
                Installaton:<br>
                </font>
                <code>
                remotes::install_github('schuyler-smith/phylosmith')
                </code>
                <br><br>
                <font size="-1">A more detailed set of installation instructions can be found <a href="https://github.com/schuyler-smith/phylosmith#installation" target="_blank">here</a>.
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/ssBLAST" target="_blank">ssBLAST</a></b></font><br><br>
                <font size="-1">An R-Package hosted on <a href="https://github.com/schuyler-smith/ssBLAST" target="_blank">Github</a>. The package contains functions for working with BLAST outputs and FATA/FASTQ files. All functions are written in c++ for fast and efficient handling of large files. The functions are written to handle both zipped and unzipped versions of sequencing files. Functions include extraction of read names from FATA files, filter/process BLAST outputs, write new FASTA/FASTQ files containing reads that aligned or did not align, and dereplicating sequences.
                <br><br>
                Installaton:<br>
                </font>
                <code>
                remotes::install_github('schuyler-smith/ssBLAST')
                </code>
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/simple.dada" target="_blank">simple.dada</a></b></font><br><br>
                <font size="-1">An R-Package hosted on <a href="https://github.com/schuyler-smith/simple.dada" target="_blank">Github</a>. The package contains functions for simplifying the DADA2 pipeline. The functions try to use read quality scores to intuitively decide paramters of the DADA2 functions. The idea is to provide a single line command for processing 16S rRNA sequencing data. The best approach is to run each step of DADA2 individually, to control the decision process. This package was made mostly for labmates to use on datasets that I knew had high-quality reads.
                <br><br>
                Installaton:<br>
                </font>
                <code>
                remotes::install_github('schuyler-smith/simple.dada')
                </code>
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/schuylR" target="_blank">schuylR</a></b></font><br><br>
                <font size="-1">An R-Package hosted on <a href="https://github.com/schuyler-smith/schuylR" target="_blank">Github</a>. This package has no focus, it simply contains functions that I repeatedly find useful.
                <br><br>
                Installaton:<br>
                </font>
                <code>
                remotes::install_github('schuyler-smith/schuylR')
                </code>
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/genome_tools" target="_blank">genome_tools</a></b></font><br><br>
                <font size="-1">An R-Package hosted on <a href="https://github.com/schuyler-smith/genome_tools" target="_blank">Github</a>. The package contains functions for various operations for working with genomic data. Many of the functions are not exported from the package, mostly stored in the repository.
                <br><br>
                Installaton:<br>
                </font>
                <code>
                remotes::install_github('schuyler-smith/genome_tools')
                </code>
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/linux_box_sync" target="_blank">linux_box_sync</a></b></font><br><br>
                <font size="-1">A BASH program hosted on <a href="https://github.com/schuyler-smith/linux_box_sync" target="_blank">Github</a>. The program is written sync a local Box directory with the cloud. I wrote this because collaborators wanted to keep all of the project files on Box, but there is no Box app for Linux. This functions makes it simple to use the Box API to copy files to and from the cloud.
                <br><br>
                Installaton: clone the repo or download/copy the <code>box_sync.sh</code> file. the shell script should run as-is. <br>
                A more detailed set of installation instructions can be found <a href="https://github.com/schuyler-smith/linux_box_sync#usage" target="_blank">here</a>.
                </font>
                </li>
                <!--  -->
                <!--  -->
                <li><font size="+2"><b><a href="https://github.com/schuyler-smith/hpcc_R_setup" target="_blank">hpcc_R_setup</a></b></font><br><br>
                <font size="-1">A BASH program hosted on <a href="https://github.com/schuyler-smith/hpcc_R_setup" target="_blank">Github</a>. The program is written to setup R on the MSU-HPCC. Setting up the R environment and using locally installed packages is non-intuitive in the HPCC system. I wrote this program to make it easy for labmates to get their R setup and use the shared package folder that I maintained.
                <br><br>
                Instructions can be found <a href="https://github.com/schuyler-smith/hpcc_R_setup#setup-r" target="_blank">here</a>.
                </font>
                </li>
            </ul>
        </div>
        </div>

<!--         -->

        <!-- MEDIA LIST -->
        <div id="medialist">
            <ul>
                {% for link in site.data.media %}
                    <li><a class="{{ link.icon }}" href="{{ link.url }}" target="_blank" ><span class="description">{{ link.title }}</span></a></li>
                {% endfor %}
            </ul>
        </div>
        <!-- END MEDIA LIST -->
</body>
</html>
